src/vm/eventpipe.cpp

```diff
@@ -503,6 +516,8 @@ void WINAPI EventPipe::FlushTimer(PVOID parameter, BOOLEAN timerFired)
     }
     CONTRACTL_END;
 
+    GCX_PREEMP();
```

Comment:
Is there a reason you moved the `GCX_PREEMP()`?

Reply:
Moved it outside the `Crst`, just like we do on `Disable`. On the other hand, @cshung is reviewing how we use the `MODE_*`s